src/frontend/src/pages/AdminToolsPage/ProjectsConfig/LinkTypeFormModal.tsx

```diff
@@ -0,0 +1,96 @@
+import { Controller, useForm } from 'react-hook-form';
+import NERFormModal from '../../../components/NERFormModal';
+import { FormControl, FormLabel, FormHelperText, Switch, Stack } from '@mui/material';
+import ReactHookTextField from '../../../components/ReactHookTextField';
+import { useToast } from '../../../hooks/toasts.hooks';
+import * as yup from 'yup';
+import { yupResolver } from '@hookform/resolvers/yup';
+import { LinkType } from 'shared';
+import { EditLinkTypePayload } from '../../../hooks/projects.hooks';
+
+interface LinkTypeFormModalProps {
+  showModal: boolean;
+  handleClose: () => void;
+  defaultValues?: LinkType;
+  onSubmit: (data: EditLinkTypePayload) => void;
```

Comment:
this might be a bit weird because edit and create have different payloads since you can't edit a link type's name, for now just do it for create, we'll figure out edit once that's done